js/base-tool-table.js

```diff
@@ -0,0 +1,385 @@
+        // For example, finds the d100 column
+        const dplace = labels.findIndex(l => /d[0-9]+/.test(l));
+        const tlen = labels.length;
+
+        r20t.tableitems.reset(t.rows.map(i => {
+            // Create the return value
+            const out = {
+                id: d20plus.ut.generateRowId(),
+                name: "",
+            };
+
+            // Set the name
+            for (let col = 0; col < tlen; col++) {
+                // Add a seperator for cases of multiple columns
+                if (out.name.length > 0) {
+                    out.name += " | "
+                }
+                // Add each column to out.name
+                if (col !== dplace) {
+                    // Get rid of ugly notation
+                    clean = i[col].replace(/\{@[\w\d]* (.*)}/, "$1");
+                    out.name += clean;
+                }
+            }
+
+            // Set the weight
+            if (dplace !== -1) {
+                const weight = i[dplace];
+                const dash = weight.indexOf("–"); // Note: – is different from -
+
+                // If the weight is a range
+                if (dash !== -1) {
+                    // Get the two numbers in the range, subtract them, add 1
+                    const low = parseInt(weight.substring(0, dash));
+                    const high = parseInt(weight.substring(dash + 1));
+                    if (high === 0) high = 100;
+                    out.weight = high - low + 1;
+                } else { // If the weight is a single value
+                    out.weight = 1;
+                }
+            } else { // If the weight is unlisted
+                out.weight = 1;
+            }
+
+            if (i.avatar) out.avatar = i.avatar;
+            return out;
+        }));
+        r20t.tableitems.forEach(it => it.save());
+    }
+
+    // A function for generating and saving tables from JSONs of the Roll20 format
+    function createR20Format(t) {
+        const r20t = d20.Campaign.rollabletables.create({
+            name: t.name.replace(/\s+/g, "-"),
+            showplayers: t.isShown,
+            id: d20plus.ut.generateRowId(),
+        });
+
+        r20t.tableitems.reset(t.items.map(i => {
+            const out = {
+                id: d20plus.ut.generateRowId(),
+                name: i.row,
+            };
+            if (i.weight !== undefined) out.weight = i.weight;
+            if (i.avatar) out.avatar = i.avatar;
+            return out;
+        }));
+        r20t.tableitems.forEach(it => it.save());
+    }
+
+    // Function to create from create your own format
+    // Note that due to the simplicity of the format, hiding from players, weight, and avatars can't be configured
+    function createCreateFormat(t) {
+        // Split the input into lines
+        const lines = t.split("\n");
+
+        // Ensure there's a table name and at least one element
+        if (lines.length < 2) {
+            return;
```

Comment:
does this alert the user in any way, or just silently fail? Because the latter would be bad